waiter/src/waiter/auth/authentication.clj

```diff
@@ -98,9 +103,14 @@
 ;;   - or upon successful authentication populate the request with :authorization/user and :authorization/principal"
 (defrecord SingleUserAuthenticator [run-as-user password]
   Authenticator
+  (process-callback [_ request scheme operation]
+    (throw (ex-info "Single user authenticator does not support callbacks."
+                    {:scheme scheme
+                     :operation operation
+                     :status 400})))
   (wrap-auth-handler [_ request-handler]
     (fn anonymous-handler [request]
-      (handle-request-auth request-handler request run-as-user run-as-user password))))
+      (handle-request-auth request-handler request run-as-user (auth-params-map run-as-user run-as-user) password))))
```

Comment:
Nitpick: I would prefer extracting `(auth-params-map run-as-user run-as-user)` into a `let` variable to make the line easier to read.

Reply:
done